packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -103,6 +108,44 @@ class BuildIOSCommand extends BuildSubCommand {
       throwToolExit('Encountered error while building for $logTarget.');
     }
 
+    if (buildInfo.codeSizeDirectory != null) {
+      final SizeAnalyzer sizeAnalyzer = SizeAnalyzer(
+        fileSystem: globals.fs,
+        logger: globals.logger,
+        processUtils: processUtils,
+        appFilenamePattern: 'App'
+      );
+      // Only support 64bit iOS code size analysis.
+      final String arch = getNameForDarwinArch(DarwinArch.arm64);
+      final File aotSnapshot = globals.fs.directory(buildInfo.codeSizeDirectory)
+        .childFile('snapshot.$arch.json');
+      final File precompilerTrace = globals.fs.directory(buildInfo.codeSizeDirectory)
+        .childFile('trace.$arch.json');
+
+      // This analysis is only supported for release builds, which also excludes the simulator.
+      // Attempt to guess the correct .app by picking the first one.
+      final Directory candidateDirectory = globals.fs.directory(
```

Comment:
What happens if users do `flutter build ios --debug --analyse-size` here and on other platforms?